### PR TITLE
Post: Announcing TypeScript support in Electron

### PR DESCRIPTION
**Pull request by zeke**
Staging URL: https://zeke.github.io/electron.atom.io/blog/2017/05/16/typescript

- [x] Update `1.6.10` if it doesn't end up being the right version number
- [x] Change the post date if we don't ship it on Tuesday May 16
- [x] ~~screencast~~ screenshots of VSC in action

This is ready for review.

cc @electron/maintainers

### CODE REVIEW

File: _posts/2017-05-16-typescript.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+of community-maintained definition files.
+
+## Installation
+
+Starting at version 1.6.10, every release of Electron includes its own
+TypeScript definition file. When you install the `electron` package from npm,
+the `electron.d.ts` file is bundled automatically with the
+installed package.
+
+The [safest way]((https://electron.atom.io/docs/tutorial/electron-versioning/)) to install Electron is using an exact version number:
```
Comment (malept) [inline _posts/2017-05-16-typescript.md]: Link is too Lisp-y, so it doesn't render.

Comment (zeke) [inline _posts/2017-05-16-typescript.md]: Fixed. Thanks.

**Comment by bpasero** [comment]
@zeke awesome! Btw I am seeing some bugs in the electron.d.ts:

image

image

image

Are you guys seeing this as well, is there a bug already?

**Comment by MarshallOfSound** [comment]
@bpasero, those have been fixed but won't see a release till the next beta.  We also introduced tests to ensure we don't publish bugged typings agajn

**Comment by bpasero** [comment]
@MarshallOfSound awesome! very happy about Electron shipping electron.d.ts btw 👍

File: _posts/2017-05-16-typescript.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+---
+title: "Announcing TypeScript support in Electron"
+author: zeke
+---
+
+The `electron` npm package now includes a TypeScript definition file that
+that provides detailed annotations of the entire Electron API. Just
```
Comment (simurai) [inline _posts/2017-05-16-typescript.md]: that ~that~

Comment (zeke) [inline _posts/2017-05-16-typescript.md]: Fixed. Thanks.

**Comment by zeke** [comment]
@bpasero if you'd be so kind as to try the latest, it would be much appreciated:

```
cd electron/electron
git checkout master
git pull
npm install
npm run create-typescript-definitions
cp out/electron.d.ts ~/your-electron-project/node_modules/electron
```

File: _posts/2017-05-16-typescript.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+TypeScript is an open-source programming language created by Microsoft. It's
+a superset of JavaScript that extends the language by adding support for
+static types. The TypeScript community has grown quickly in recent years,
+and TypeScript was ranked among the
+[most loved programming languages](https://stackoverflow.com/insights/survey/2017#technology-most-loved-dreaded-and-wanted-languages)
+in a recent Stack Overflow developer survey.
+
+TypeScript supports many of the newer language features in JavaScript like
+classes, object destructuring, and async/await, but its real differentiating
+feature is _type annotation_.
```
Comment (DanielRosenwasser) [inline _posts/2017-05-16-typescript.md]: annotation**s**

Comment (zeke) [inline _posts/2017-05-16-typescript.md]: I was waffling on that plural. You've tipped the scales.

**Comment by zeke** [comment]
This post is ready to go. We can ship it once the 1.6 release line is out of beta.

cc @kevinsawicki

**Comment by zeke** [comment]
🚀 

https://electron.atom.io/blog/2017/06/01/typescript

https://twitter.com/electronjs/status/870317132151521285

**Pull request merged**